libs/backend/src/scan/cast_vote_records/export.ts

```diff
@@ -0,0 +1,296 @@
+                : undefined,
+          },
+          {
+            interpretation: back,
+            inlineBallotImage:
+              backHasWriteIns && imageOptions.includeInlineBallotImages
+                ? await getInlineBallotImage(backFilename)
+                : undefined,
+            imageFileUri:
+              imageOptions.includedImageFileUris === 'all' ||
+              (imageOptions.includedImageFileUris === 'write-ins' &&
+                backHasWriteIns)
+                ? `file:./${imageOptions.imagesDirectory}/${backFilename}`
+                : undefined,
+          },
+        ],
+        ballotPageLayoutsLookup,
+      });
+    }
+  }
+}
+
+async function* concatStreams(readables: NodeJS.ReadableStream[]) {
+  for (const readable of readables) {
+    for await (const chunk of readable) {
+      yield chunk;
+    }
+  }
+}
+
+/**
+ * JSON format requires that we have no trailing commas, so we must
+ * post-process the cast vote records iterator to add a comma after all but
+ * the last.
+ */
+async function* commaSeparateStringGenerator(
+  asyncGenerator: AsyncGenerator<string>
+): AsyncGenerator<string> {
+  let previous: Optional<string>;
+  for await (const current of asyncGenerator) {
+    if (previous) {
+      yield `${previous},`;
+    }
+    previous = current;
+  }
+
+  // yield last element without trailing comma
+  if (previous) {
+    yield previous;
+  }
+}
+
+interface GetCastVoteRecordReportStreamParams
+  extends GetCastVoteRecordGeneratorParams {
+  ballotsCounted: number;
+  isTestMode: boolean;
+  batchInfo: BatchInfo[];
+}
+
+/**
+ * Returns a readable stream consisting of a cast vote record report.
+ */
+export function getCastVoteRecordReportStream({
+  election,
+  electionId,
+  scannerId,
+  definiteMarkThreshold,
+  isTestMode,
+  ballotPageLayoutsLookup,
+  resultSheetGenerator,
+  batchInfo,
+  imageOptions,
+}: GetCastVoteRecordReportStreamParams): NodeJS.ReadableStream {
+  const castVoteRecordGenerator = getCastVoteRecordGenerator({
+    election,
+    electionId,
+    scannerId,
+    definiteMarkThreshold,
+    ballotPageLayoutsLookup,
+    resultSheetGenerator,
+    imageOptions,
+  });
+  const reportMetadata = buildCastVoteRecordReport({
+    election,
+    electionId,
+    generatingDeviceId: scannerId,
+    scannerIds: [scannerId],
+    reportTypes: [CVR.ReportType.OriginatingDeviceExport],
+    isTestMode,
+    batchInfo,
+  });
+
+  const reportMetadataJson = JSON.stringify(reportMetadata, undefined, 2);
+
+  const reportMetadataStream = Readable.from(
+    reportMetadataJson.replace(/\n\}$/, ',\n  "CVR": [')
+  );
+
+  const cvrStream = Readable.from(
+    commaSeparateStringGenerator(
+      mapAsync(castVoteRecordGenerator, (cvr) => `\n    ${JSON.stringify(cvr)}`)
+    )
+  );
+
+  const closingStream = Readable.from('  ]\n}');
+
+  return Readable.from(
+    concatStreams([reportMetadataStream, cvrStream, closingStream])
+  );
```

Comment:
hmm. thinking about this

Reply:
I will write this if I find myself reimplementing it in different places (even in CVR fixture generator, it looks a bit different) but leave it for now.

I actually wonder if we need to be using a generator at all. Once we take out the inline ballot images, the string of an entire file containing thousands of CVRs should safely fit within a string and we could just `JSON.stringify` things.

Reply:
I'm not opposed to simplifying to remove streaming

Reply:
With the inline ballot images it would break everything, but in the next PR that changes the inline ballot images I will look into it and see if I can find the threshold. If we can export say 10,000 CVRs without problems then I'd rather just stringify it.